actionpack/lib/action_dispatch/system_testing/test_helpers/screenshot_helper.rb

```diff
@@ -0,0 +1,57 @@
+module ActionDispatch
+  module SystemTesting
+    module TestHelpers
+      # Screenshot helper for system testing
+      module ScreenshotHelper
+        # Takes a screenshot of the current page in the browser.
+        #
+        # +take_screenshot+ can be used at any point in your system tests to take
+        # a screenshot of the current state. This can be useful for debugging or
+        # automating visual testing.
+        def take_screenshot
+          save_image
+          puts "[Screenshot]: #{image_path}"
+          puts display_image
+        end
+
+        # Takes a screenshot of the current page in the browser if the test
+        # failed.
+        #
+        # +take_failed_screenshot+ is included in <tt>system_test_helper.rb</tt> that is
+        # generated with the application. To take screenshots when a test fails
+        # add +take_failed_screenshot+ to the teardown block before clearing
+        # sessions.
+        def take_failed_screenshot
+          take_screenshot unless passed?
+        end
```

Comment:
I think that would be super cool, but we'd have to hook into Capybara to do that. Maybe I can find a way to include it underneath? I'll have to look at the test runner, and since this won't change the public API it's not a merge blocker.